app/email/email_batching.go

```diff
@@ -166,47 +204,34 @@ func (job *EmailBatchingJob) checkPendingNotifications(now time.Time, handler fu
 				inspectedTeamNames[notification.teamName] = team.Id
 			}
 
-			// if the user has viewed any channels in this team since the notification was queued, delete
-			// all queued notifications
 			channelMembers, err := job.service.store.Channel().GetMembersForUser(inspectedTeamNames[notification.teamName], userID)
 			if err != nil {
 				mlog.Error("Unable to find ChannelMembers for user", mlog.Err(err))
 				continue
 			}
 
+			deleted := false
 			for _, channelMember := range channelMembers {
 				if channelMember.LastViewedAt >= batchStartTime {
```

Comment:
I guess I am just trying to understand the existing logic but it seems like we are always comparing with the time of the first notification? Is that expected?

Reply:
That seems like it. Don't ask me why :man_shrugging: :joy: